src/java.xml/share/classes/javax/xml/xpath/XPathFactory.java

```diff
@@ -369,4 +369,59 @@ public abstract boolean getFeature(String name)
     * @return New instance of an <code>XPath</code>.
     */
     public abstract XPath newXPath();
+
+    /**
+     * Sets a property for this {@code XPathFactory} and {@code XPath}
+     * created by this factory.
+     *
+     * @implSpec
+     * The default implementation in the XPath API throws
+     * {@link java.lang.UnsupportedOperationException}.
+     *
+     * @param name the property name
+     * @param value the value for the property
+     *
+     * @throws XPathFactoryConfigurationException if the property name is not
+     * recognized, or the value can not be assigned
+     * @throws UnsupportedOperationException if the implementation does not
+     * support the method
+     * @throws NullPointerException if the {@code name} is {@code null}.
+     *
+     * @since 18
+     */
+    public void setProperty(String name, String value)
+        throws XPathFactoryConfigurationException {
+
+        if (name == null) {
+            throw new NullPointerException("the name parameter is null");
+        }
+        throw new UnsupportedOperationException("not implemented");
+    }
+
+    /**
+     * Returns the value of the specified property.
+     *
+     * @implSpec
+     * The default implementation in the XPath API throws
+     * {@link java.lang.UnsupportedOperationException}.
+     *
+     * @param name the property name
+     * @return the value of the property.
+     *
+     * @throws XPathFactoryConfigurationException if the property name is not
+     * recognized
```

Comment:
Are you sure XPathFactoryConfigurationException is the right exception here? This exception suggests "a configuration error in a XPathFactory environment". I would expect something like IllegalArgumentException here.

Reply:
Agree, IAE would have been more appropriate. The only reason I went with XCE was consistency among the XPath API. It's unfortunate it has not been consistent among the XML libs. The DOM/StAX/XSLT went with IAE,  SAX/Validation SAXNotRecognizedException and SAXNotSupportedException, and then XPath XCE which was designed for config error. But if we don't plan to fix XPathFactory::set/getFeature, should we go with sth different for the property methods?

Reply:
Roger also suggested changing to IAE. I've just pushed a change to IAE. Thanks.